apps/api/src/chain/services/block-http/block-http.service.ts

```diff
@@ -2,14 +2,23 @@ import { BlockHttpService as BlockHttpServiceCommon } from "@akashnetwork/http-s
 import { singleton } from "tsyringe";
 
 import { Memoize } from "@src/caching/helpers";
+import { BlockRepository } from "@src/chain/repositories/block.repository";
 import { averageBlockTime } from "@src/utils/constants";
 
 @singleton()
 export class BlockHttpService {
-  constructor(private readonly blockHttpService: BlockHttpServiceCommon) {}
+  constructor(
+    private readonly blockHttpService: BlockHttpServiceCommon,
+    private readonly blockRepository: BlockRepository
+  ) {}
 
   @Memoize({ ttlInSeconds: averageBlockTime })
   async getCurrentHeight() {
     return await this.blockHttpService.getCurrentHeight();
   }
+
+  @Memoize({ ttlInSeconds: averageBlockTime })
+  async getLatestProcessedHeight() {
```

Comment:
This service is intended to serve as an abstraction over chain rest interface for blocks as suggested by its name. Other http services do the same for other domains. Db dep is breaking the boundaries of concerns here

Reply:
Ok I'll use the repo directly in the cleaner service as you suggested.